balance.py

```diff
@@ -497,27 +510,67 @@ def callback_teams(self, players, channel):
 
     def cmd_do(self, player, msg, channel):
         """Forces a suggested switch to be done."""
-        if self.suggested_pair:
-            self.execute_suggestion()
+        if not self.suggested_pair:
+            return
+        
+        if not self.switch_opt_in:
+            return
+
+        self.execute_suggestion()
+
+    def cmd_dont(self, player, msg, channel):
+        """Vetoes a suggested switch to be done."""
+        if not self.suggested_pair:
+            return
+
+        if self.switch_opt_in:
+            return
+
+        self.msg("An admin prevented the switch! The switch will be terminated.")
+        self.suggested_pair = None
+        self.suggested_agree = [False, False]
 
     def cmd_agree(self, player, msg, channel):
         """After the bot suggests a switch, players in question can use this to agree to the switch."""
-        if self.suggested_pair and not all(self.suggested_agree):
-            p1, p2 = self.suggested_pair
+        if not self.suggested_pair or not self.switch_opt_in or all(self.suggested_agree):
+            return
+
+        p1, p2 = self.suggested_pair
             
-            if p1 == player:
-                self.suggested_agree[0] = True
-            elif p2 == player:
-                self.suggested_agree[1] = True
-
-            if all(self.suggested_agree):
-                # If the game's in progress and we're not in the round countdown, wait for next round.
-                if self.game.state == "in_progress" and not self.in_countdown:
-                    self.msg("The switch will be executed at the start of next round.")
-                    return
+        if p1 == player:
+            self.suggested_agree[0] = True
+        elif p2 == player:
+            self.suggested_agree[1] = True
 
-                # Otherwise, switch right away.
-                self.execute_suggestion()
+        if not all(self.suggested_agree):
+            return
+
+        # If the game's in progress and we're not in the round countdown, wait for next round.
+        if self.game.state == "in_progress" and not self.in_countdown:
+            self.msg("The switch will be executed at the start of next round.")
+            return
+
+        # Otherwise, switch right away.
+        self.execute_suggestion()
+
+    def cmd_veto(self, player, msg, channel):
+        """After the bot suggests a switch, players in question can use this to veto to the switch."""
+        if not self.suggested_pair or self.switch_opt_in:
+            return
```

Comment:
Similar as in https://github.com/MinoMino/minqlx-plugins/pull/28/files#r672927389